Sources/BranchSDK/BNCServerRequestQueue.m

```diff
@@ -287,6 +287,8 @@ - (void)retrieve {
                 }
             }
             self.queue = decodedQueue;
+            // Requests are loaded into queue now. Delete queue file stored on disk.
+            [self removeSaveFile];
```

Comment:
Is this line always executed at the end? In other words, can an error scenario prevent this line?

Reply:
In current implementation, retrive function is called only once (called using function dispatch_once - singleton class). So no threading issues I see here.
This line will not be executed only if `data` is empty or any error happened while reading the cache file. That is correct behavior.
Other than that, I dont see anything stopping this line execution after decoded queue is assigned to SDK request queue `self.queue = decodedQueue;`